### PR TITLE
[image_picker] Bump minimum Flutter version and iOS deployment target

### PR DESCRIPTION
**Pull request by jmagman**
Flutter apps run on 2.4.0-0.0.pre and later [will be upgraded to a minimum of iOS 9.0](https://github.com/flutter/flutter/pull/85174).  Now that Flutter 2.5 has hit stable and [iOS 8 support has been dropped](https://flutter.dev/go/rfc-ios8-deprecation), change the plugin minimum iOS version to `9.0`.  

Bump the Flutter and dart version constraints to 2.5 and 2.14 respectively to enforce that the iOS 9.0 migration has happened on the app side, because an older 8.0 app won't build with a 9.0 plugin.  I believe the last time this was done was a few months ago for null safety adoption https://github.com/flutter/plugins/pull/3324 (there's recent precedent).

Users on Flutter versions lower than the current stable 2.5 will not be able to upgrade this plugin past this version.  

image_picker part of https://github.com/flutter/flutter/issues/84198

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/image_picker/image_picker/example/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -684,7 +684,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
```
Comment (jmagman) [inline packages/image_picker/image_picker/example/ios/Runner.xcodeproj/project.pbxproj:687]: Same, done automatically via https://github.com/flutter/flutter/pull/85174.

File: packages/image_picker/image_picker/example/ios/Flutter/AppFrameworkInfo.plist [hunk 2]
```diff
@@ -25,6 +25,6 @@
     <string>arm64</string>
   </array>
   <key>MinimumOSVersion</key>
-  <string>8.0</string>
+  <string>9.0</string>
```
Comment (jmagman) [inline packages/image_picker/image_picker/example/ios/Flutter/AppFrameworkInfo.plist:28]: This was done automatically via https://github.com/flutter/flutter/pull/85174 by just `flutter build`ing the example project.

File: packages/image_picker/image_picker/pubspec.yaml [hunk 3]
```diff
@@ -3,11 +3,11 @@ description: Flutter plugin for selecting images from the Android and iOS image
   library, and taking new pictures with the camera.
 repository: https://github.com/flutter/plugins/tree/master/packages/image_picker/image_picker
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+image_picker%22
-version: 0.8.4
+version: 0.8.5
```
Comment (cyanglaz) [inline packages/image_picker/image_picker/pubspec.yaml:6]: similar to the google map PR, 0.8.4+1 is probably enough.

Comment (jmagman) [reply packages/image_picker/image_picker/pubspec.yaml]: Done.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Comment by nguyenlamzx** [comment]
@jmagman can we keep the previous sdk constrains? because there're many packages didn't update to sdk and I using this package along with other packages

**Comment by jmagman** [comment]
> @jmagman can we keep the previous sdk constrains? because there're many packages didn't update to sdk and I using this package along with other packages.
> 
> For now, I use old version `image_picker: 0.8.4+1` to be able to install successfully

From https://github.com/flutter/plugins/commit/701fdb6eab07c99d83fe583f1b7630100fe01a0e#r56918265

> Flutter 2.5 shipped with Dart 2.14, so that's the minimum of the range. You can see in [flutter/flutter#84198](https://github.com/flutter/flutter/issues/84198) why we wanted to target 2.5, namely we need to make sure all consumers of the plugin have upgraded their apps to iOS 9 or later before this plugin could drop support for iOS 8.
> If you're seeing an issue related to this bump, please file a new issue so the ecosystem team can evaluate this strategy.

Or you can comment on https://github.com/flutter/flutter/issues/84198

**Pull request merged**